unicorn/src/service/unicorn.cpp

```
@@ -67,7 +68,12 @@ class unicorn_slot_t :
             try {
                 result.write(future.get());
             } catch (const std::system_error& e) {
-                result.abort(e.code(), error::to_string(e));
+                COCAINE_LOG_WARNING(service->log, "could not assign to deferred - {}", error::to_string(e));
+                try {
+                    result.abort(e.code(), error::to_string(e));
+                } catch(const std::system_error& e2) {
+                    COCAINE_LOG_WARNING(service->log, "could not abort deferred - {}", error::to_string(e2));
```

I doubt that swallowing an exception is better